File: spark-python/pyspark-sql/src/sql/metastore/spark/spark_metastore.py
import os

from pyspark.sql import SparkSession

if __name__ == '__main__':

    warehouse_location = os.environ["SPARK_WAREHOUSE"]
    derby_home = os.environ["derby.system.home"]

    spark = (SparkSession.builder
             .appName("spark_hive_metastore")
             .config("spark.sql.catalogImplementation", "in-memory")
             .config("spark.driver.extraJavaOptions",
                     f"-Dderby.system.home='{derby_home}'")
             .enableHiveSupport()
             .getOrCreate())

File: spark-scala/spark-sql/spark-sql-hive/src/main/scala/com/mahfooz/spark/sqlhive/metastore/SparkMetastore.scala
package com.mahfooz.spark.sqlhive.metastore

import org.apache.spark.sql.SparkSession

object SparkMetastore {

  def main(args: Array[String]): Unit = {
    val sparkWarehouse = sys.env.getOrElse("SPARK_WAREHOUSE","spark-warehouse")

    println(sparkWarehouse)

    System.setProperty("derby.system.home", sys.env.getOrElse("derby.system.home",""))

    val spark = SparkSession
      .builder()
      .appName("ShowAllPartitions")
      .master("local[*]")
      .config("spark.sql.warehouse.dir", sparkWarehouse)
      .enableHiveSupport()
      .getOrCreate()

    println(spark.version)
    println(s"spark.sql.catalogImplementation = ${spark.conf.get("spark.sql.catalogImplementation")}")
    println(s"metastore.catalog.default = ${spark.conf.get("metastore.catalog.default")}")
  }

}
